### PR TITLE
Fix issues related to generating a FlowNet from 2D input

### PR DESCRIPTION
**Pull request by olelod**
When all completions/wells are at the same depth, the Delauney triangulation can be performed in 2D and requires only 4 points (5 points in 3D). This PR fixes the algorithm so it works for such a problem as well.

---

### Contributor checklist

- [ ] :tada: This PR closes #ISSUE_NUMBER.
- [ ] :scroll: I have broken down my PR into the following tasks:
   - [ ] Task 1
   - [ ] Task 2
- [ ] :robot: I have added tests, or extended existing tests, to cover any new features or bugs fixed in this PR.
- [ ] :book: I have considered adding a new entry in `CHANGELOG.md`.
- [ ] :books: I have considered updating the documentation.

### CODE REVIEW

File: src/flownet/network_model/_generate_connections.py [hunk 1]
```diff
@@ -31,8 +31,11 @@ def _is_angle_too_large(
     """
     calculate_angle = np.rad2deg(
         math.acos(
-            (math.pow(side_a, 2) + math.pow(side_b, 2) - math.pow(side_c, 2))
-            / (2 * side_a * side_b)
+            min(
```
Comment (olelod) [inline src/flownet/network_model/_generate_connections.py:34]: Have observed issued due to round off error like trying math.acos(1.0000002)

**Comment by olwijn** [comment]
I ran an example, looks good.

**Review by olwijn — COMMENTED** [review]
I ran an example, looks good.

**Review by fahaddilib — APPROVED** [review]
LGTM

**Pull request merged**